File: Data Structures and Algorithms/Sword refers to offer/C++/11.cpp
//1
class Solution {
public:
     int  NumberOf1(int n) {
	 bitset<32> bit(n);
	 return bit.count();    	
     }
};

//2
class Solution {
public:
     int  NumberOf1(int n) {
        int count=0;
        while(n)
        {
        	count++;
        	n=n&(n-1);
        }
        return count;
     }
};

//3
class Solution {
public:
     int  NumberOf1(int n) {
        int count=0;
        int i=0;
        unsigned int flag;
        while(flag!=0)
        {
			if((n&flag)!=0)
			{
				count++;
			}
			flag<<=1;
        }
        return count;
     }
};

File: Data Structures and Algorithms/Sword refers to offer/Python/11.py
# -*- coding:utf-8 -*-
class Solution:
    def NumberOf1(self, n):
        # write code here
        count=0
        MAX_INT=(1<<31)-1
        while n:
            if n < - MAX_INT- 1 or n > MAX_INT:
                break
                
            count+=1
            n=n&(n-1)
        return count
